lib/ansible/modules/packaging/os/zypper.py

```diff
@@ -107,6 +108,18 @@
         required: false
         default: "no"
         choices: [ "yes", "no" ]
+    retries:
+        version_added: "2.3"
+        description:
+            - Number of retries if another instance of zypper is running. Defaults to no retries.
+        required: false
+        default: 0
+    retry_interval:
+        version_added: "2.3"
+        description:
+            - Time in seconds to wait between retries.
+        required: false
+        default: 10
```

Comment:
many API modules also have this built in, i'm fine with modules doing this internally when dealing with notably unreliable services (or locking thundering herds). As an explicit option I prefer to keep things simple and point people to the until directive.

If we want to make it easier for people we can document the rc codes in the module docs (and add until example?)